pkg/config/mail/mailgunImpl.go

```diff
@@ -0,0 +1,73 @@
+package mail
+
+import (
+	"context"
+	"fmt"
+	"github.com/joho/godotenv"
+	"github.com/mailgun/mailgun-go/v3"
+	"os"
+	"time"
+)
+//func INewService(repository irepository.IUserRepository) iservice.IUserService {
+//	return service{repository}
+//}
+func SendSimpleMessage(domain, apiKey string) (string, error) {
+	mg := mailgun.NewMailgun(domain, apiKey)
+	m := mg.NewMessage(
+		"Excited User <mailgun@YOUR_DOMAIN_NAME>",
+		"Hello",
+		"Testing some Mailgun awesomeness!",
+		"YOU@YOUR_DOMAIN_NAME",
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	_, id, err := mg.Send(ctx, m)
+	return id, err
+}
+
+
+type mailGunImplementation struct {
+	domain string
+	apiKey string
+
+}
+
+
+func NewMailGunImplementationNoArgs() IMail {
```

Comment:
I'm not sure if there's method overloading in go but if there is then `NewMailGunImplementation` and `NewMailGunImplementationNoArgs ` can have the same name but take different parameters

Reply:
actually no method overloading.. actually NewMailGunImplementation for the purpose of testing and if you wanted to inject a different parameter
NewMailGunImplementationNoArgs=>to be used on the fly

Reply:
Okay, thanks